lib/puppet/provider/service/systemd.rb

```diff
@@ -34,6 +34,22 @@ def self.instances
     return []
   end
 
+  # Static services cannot be enabled or disabled manually. Indirect services 
+  # should not be enabled or disabled due to limitations in systemd (see 
+  # https://github.com/systemd/systemd/issues/6681).
+  def enabled_insync?(current)
+    case cached_enabled?
+    when 'static'
+      Puppet.debug("Unable to enable or disable static service #{@resource[:name]}")
+      return true
+    when 'indirect'
+      Puppet.warning("Service #{@resource[:name]} is in 'indirect' state and cannot be enabled/disabled")
```

Comment:
One more question, when `static` we use debug, for indirect we use `warning`. Won't this warning generate too much noise for the users? Why not use debug here also?

Reply:
Before this commit, users were allowed to try enabling/disabling `indirect` services, using puppet, but as the new state cannot be checked at the next run because of systemd limitations, changes were being reported at every run (as `indirect` services are considered disabled in puppet and users were trying to enable them). 
Because of this it made sense to me to have the log set to `warning` (to let users know that they shouldn't enable/disable such services) but I agree that this can produce noise and I agree debug level would be better (also better from a consistency point of view).

Reply:
Before this commit, users were allowed to try enabling/disabling `indirect` services, using puppet, but as the new state cannot be checked at following runs because of systemd limitations, changes were being reported at every time (as `indirect` services are considered `disabled` in puppet and users were trying to `enable` them). 

Because of the issue itself, it made sense to me to have the log set to `warning` (trade idempotency issue with a warning to let users know that they shouldn't enable/disable such services) but I can see how this can produce unwanted noise and I agree `debug` level would be better (from a consistency point of view also).